src/main/java/org/apache/sling/commons/threads/impl/DefaultThreadPoolManager.java

```diff
@@ -404,23 +426,56 @@ public ThreadPoolExecutor getExecutor() {
             return null;
         }
 
-        protected void unregisterMBean() {
+        protected void unregisterMBeanAndMetrics() {
             if ( this.mbeanRegistration != null ) {
                 this.mbeanRegistration.unregister();
                 this.mbeanRegistration = null;
             }
+            unregisterGauges();
         }
 
-        protected void registerMBean() {
+        protected void registerMBeanAndMetrics() {
             try {
                 final Dictionary<String, String> mbeanProps = new Hashtable<String, String>();
                 mbeanProps.put("jmx.objectname", "org.apache.sling:type=threads,service=ThreadPool,name=" + this.name);
 
                 final ThreadPoolMBeanImpl mbean = new ThreadPoolMBeanImpl(this);
                 this.mbeanRegistration = bundleContext.registerService(ThreadPoolMBean.class.getName(), mbean, mbeanProps);
+
+                // not exposing MaxThreadAge as bean.getMaxThreadAge() is deprecated due
+                // to SLING-6261 and always returns -1
+                String[] ignoredAttributes = { "MaxThreadAge" };
+
+                for(Map.Entry<String, Gauge<?>> nameToGauge : ThreadPoolMetricsGauges.create(mbean, mbean.getMBeanInfo(),
+                        Arrays.asList(ignoredAttributes)).entrySet()) {
+                    addGauge(nameToGauge.getKey(), nameToGauge.getValue());
+                }
             } catch (Throwable t) {
                 logger.warn("Unable to register Thread Pool MBean", t);
             }
         }
+
+        private void addGauge(String name, Gauge<?> gauge) {
+            final String fullName = getMetricFullName(name);
+            final Dictionary<String, Object> props = new Hashtable<>();
+            props.put(Gauge.NAME, fullName);
+            props.put(Constants.SERVICE_DESCRIPTION, "Apache Sling Metrics Gauge - " + fullName);
+            props.put(Constants.SERVICE_VENDOR, "The Apache Software Foundation");
+
+            try {
+                gaugeRegistrations.add(bundleContext.registerService(Gauge.class, gauge, props));
+            } catch (Throwable t) {
+                logger.warn("Unable to register Metrics Gauge - " + fullName, t);
+            }
```

Comment:
Please use placeholders ( `{}`) here

Reply:
For warn() (and error()) with Throwable argument specifically unfortunately there is no special version of this method that accepts both Throwable (and treating it specifically for exception presentation) and additional arguments - fullName is a local final variable and the concatenation here shouldn't cause much trouble apart from unaesthetic looks. Question whether to remove Throwable specific call and threat is just as an ordinary parameter e.g. using getMessage() for it to look better and consistent with other calls but remove the potentially specific treatment of thrown stack trace displayed by the logging end point at the and and so on?

Reply:
I think that should work, have you tried `logger.warn("Unable to register Metrics Gauge - {}" ,fullName, e);` ? At least according to https://www.baeldung.com/slf4j-log-exceptions it should be ok

Reply:
@rombert Right! I wouldn't expect it from the javadocs:
https://github.com/qos-ch/slf4j/blob/v_1.7.6/slf4j-api/src/main/java/org/slf4j/Logger.java#L531
but confirmed when I saw this:
https://github.com/qos-ch/slf4j/blob/v_1.7.6/slf4j-api/src/main/java/org/slf4j/helpers/MessageFormatter.java#L182
So a fix is there: https://github.com/apache/sling-org-apache-sling-commons-threads/pull/1/commits/6686fbf4f48529d2eb838dcf40466f1aedfa621e